sourced/ml/cmd/merge_coocc.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import operator
+from uuid import uuid4
+
+from modelforge.progress_bar import progress_bar
+import numpy as np
+from scipy.sparse import coo_matrix
+
+from sourced.ml.cmd.args import handle_input_arg
+from sourced.ml.extractors.helpers import filter_kwargs
+from sourced.ml.models import OrderedDocumentFrequencies, Cooccurrences
+from sourced.ml.transformers import CooccModelSaver
+from sourced.ml.utils.engine import pause
+from sourced.ml.utils.spark import create_spark
+
+
+MAX_INT32 = 2**31 - 1
+
+
+@pause
+def merge_coocc_entry(args):
+    log = logging.getLogger("merge_coocc")
+    log.setLevel(args.log_level)
+    filepaths = list(handle_input_arg(args.input, log))
+    log.info("Found %d files", len(filepaths))
+    df = OrderedDocumentFrequencies().load(args.docfreq)
+    if args.no_spark:
+        merge_coocc_entry_no_spark(df, filepaths, log, args)
+    else:
+        merge_coocc_entry_spark(df, filepaths, log, args)
+
+
+def merge_coocc_entry_spark(df, filepaths, log, args):
+    session_name = "merge_coocc-%s" % uuid4()
+    session = create_spark(session_name, **filter_kwargs(args.__dict__, create_spark))
+    spark_context = session.sparkContext
+    global_index = spark_context.broadcast(df.order)
+
+    coocc_rdds = []
+    for path in progress_bar(filepaths, log):
+        coocc = Cooccurrences().load(path)
+        rdd = coocc.matrix_to_rdd(spark_context)  # rdd structure: ((row, col), weight)
+        tokens = spark_context.broadcast(coocc.tokens)
+        coocc_rdds.append(
+            rdd.map(lambda row: ((global_index.value.get(tokens.value[row[0][0]], -1),
+                                  global_index.value.get(tokens.value[row[0][1]], -1)),
```

Comment:
`tuple(global_index.value.get(tokens.value[row[0][i]], -1) for i in (0, 1))`

Reply:
It would also help to have a comment here which explains this monster's structure or gives an example